### DISCUSSION TITLE
sap_hana_install: Separate major actions performed by the role

### DISCUSSION THREAD

**Discussion by berndfinger** [sap-linuxlab/community.sap_install#836] [Ideas]
For easier integration into different workflows, for faster testing and developing, certain steps of the sap_hana_install role should be separated. There should be role parameters for executing only certain steps, e.g. just create the hdblcm configfile without any initial preparation steps, just execute the SAP HANA installation, and so forth.
Which additional steps should be separated in addition to the above mentioned ones?

Do we also want separate roles for certain steps, e.g.:
- preparation/expansion of software (e.g. perform steps of sap_install_media_detect?)
- user management actions
- creation of the hdblcm configfile
- launching the hdblcm command
- certain post-install actions (opening firewall ports, drop/create tenants, ...)

In that case, we need to discuss all required interfaces (parameters) and dependencies.

**Comment by marcelmamula** [comment]
Idempotency in https://github.com/sap-linuxlab/community.sap_install/pull/1111 along with existing tags allows this separation already, so it would be mainly about adding more tags for new cases.

As per examples you mentioned:
- preparation/expansion of software (e.g. perform steps of sap_install_media_detect?)
  - Preparation will fire always for new system installation, when it is not installed already.
  - It is further limited to it with `sap_hana_install_preinstall`
  - Software handling should be left to `sap_install_media_detect`

- user management actions
  - I dont see any pre-install user steps and post install are handled under detection as we already marked them as too impactful to run on existing system.

- creation of the hdblcm configfile
  - I dont see dedicated tag, so maybe new tag might be needed.

- launching the hdblcm command
  - too dangerous to even consider it as it can break existing systems if it does not have robust validation and idempotency.

- certain post-install actions (opening firewall ports, drop/create tenants, ...)
  - Split during idempotency PR and variables are already exposed to control what is being executed.
